### PR TITLE
Issue/5136 halt UI

### LINKED ISSUE #5136: Block UI until environment is halted
https://inmanta.slack.com/archives/CKRF0C8R3/p1695028344306179

The idea would be to block the UI, may be with a spinner for example, when the user clicks the halt button. The screen should be blocked until we get a response from the halt api-call.

### PR DESCRIPTION
**Pull request by matborowczyk**
# Description

* Short description here *

closes #5136 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Code is clear and sufficiently documented
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

File: src/Core/Contracts/ManagerResolver.ts [hunk 1]
```diff
@@ -1,3 +1,13 @@
-export interface ManagerResolver<Type> {
-  get(): Type[];
+import { CommandManager } from "./CommandManager";
+import { QueryManager } from "./QueryManager";
+
+interface ICommandManagerResolver {
```
[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/Core/Contracts/ManagerResolver.ts:4]: I had to make an exception of usual style as `CommandManagerResolver` is being implemented and exported already and got errors from lint. I propose going for `CommandManagerResolverInterface`, or changing actual `CommandManagerResolver` to `CommandManagerResolverImpl`

[the remark this answers is not in the document]
Comment (LukasStordeur) [reply src/Core/Contracts/ManagerResolver.ts:4]: But this one's only used to halt the queries, so, I find it rather confusing if it has nearly the same name. It doesn't have the same implementation either, I would rather keep it line so we can understand it when we come back to it.

[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/Core/Contracts/ManagerResolver.ts:4]: I probably didn't express myself clearly, functionality for halting queries had to be done at the root of QueryManager, so there is no other QueryManagerResolver.

the interfaces for Command & Query Manager Resolvers were mergeed together as there was no differences, but from now on they are different so I had to split them and add meaningful name to that that is in line with what we use, but in the process `CommandManagerResolver ` name conflicts with name of implementation of the class with the same name, as both are exported and reused in different places.

[the remark this answers is not in the document]
Comment (LukasStordeur) [reply src/Core/Contracts/ManagerResolver.ts:4]: I'd go with what is already used. CommandManagerResolverImpl

**Review by LukasStordeur — APPROVED** [review]

**Pull request closed without merging**